run_test.sh

```diff
@@ -1541,6 +1543,75 @@ EOF
 echo 'Removing /usr/src/dkms_build_exclusive_test-1.0'
 rm -r /usr/src/dkms_build_exclusive_test-1.0
 
+############################################################################
+### Testing os-release detection                                         ###
+############################################################################
+echo "Backing up /etc/os-release and /usr/bin/os-release"
+# Get the contents of the os-release file. We do it first so symlinks resolve
+osrelease_contents=$(for f in /etc/os-release /usr/lib/os-release; do
+    if [ -f "$f" ]; then
+        cat "$f"
+    fi
+done)
+osrelease_cleanup() {
+    mv _etc-os-release /etc/os-release &>/dev/null || :
+    mv _usrlib-os-release /usr/lib/os-release &>/dev/null || :
+}
+trap osrelease_cleanup EXIT
+mv /etc/os-release _etc-os-release &>/dev/null || :
+mv /usr/lib/os-release _usrlib-os-release &>/dev/null || :
+
+echo "Adding the dkms_osrelease_stub-1.0 module with no os-release files (expected error)"
+run_with_expected_error 4 dkms add test/dkms_osrelease_stub-1.0 << EOF
+Error! System is missing os-release file.
+EOF
+
+echo "Creating /etc/os-release"
+cat > /etc/os-release <<< "$osrelease_contents"
```

Comment:
You're right! After all these years, my misunderstanding of `cp` was: if it encountered a symlink to copy, it would copy the symlink, not the backing target the symlink resolves to. I have change it to use cp :)

Reply:
I have pushed changes that fixes this. It would be nice to either just copy `_etc-os-release` or `_usrlib-os-release` instead of creating a new, `_os-release` file, but if `/etc/os-release` and `/usr/lib/os-release` are both relative symlinks, then this will fail. Because at the point where we need to copy, say, `_etc-os-release` back to `/etc/os-release`, the symlink will no longer properly resolve because it is in a different directory.